### ISSUE TITLE
Where is SpecAugment applied to transformer s2s models?

### ISSUE THREAD

**Issue by lunixbochs** [facebookresearch/wav2letter#520]
The docs and paper both reference SpecAugment:
> The model is trained with total batch size 256 for 16 epochs with Adagrad. There is a warmup stage: SpecAugment is activated only after warmup, and the learning rate is warmed up (linearly increased) over the first 64000 updates to 0.03. It is then divided by 2 at epoch 3. At epoch 8 learning rate is set to --lr=0.0075, and at epoch 14 - --lr=0.005.

But I don't see an SAUG layer in [the arch](https://github.com/facebookresearch/wav2letter/blob/master/recipes/models/sota/2019/am_arch/am_transformer_s2s.arch), and none of the flags seem SpecAugment related.

**Comment by tlikhomanenko** [comment]
Hi @lunixbochs,

So right now warm-up is not supported and to do our training we changed the Train.cpp to have it. To use SpecAug with warm-up training we didn't use the layer in arch but directly did augmentation inside code before running forward pass. 

Right now we are considering ways to release warm-up support and SpecAug usage with it. As we agree on the interface we will release the code.
